bpo-41905: added abc.update_abstractmethods (#22485)
python/cpython: Doc/library/abc.rst

   .. versionadded:: 3.4

.. function:: update_abstractmethods(cls)
   A function to recalculate an abstract class's abstraction status. This
   function should be called if a class's abstract methods have been
   implemented or changed after it was created. Usually, this function should

>    be called from within class decorator.

Reviewer: Should be "within a class decorator" (add the word "a").

   .. versionadded:: 3.4

.. function:: update_abstractmethods(cls)
   A function to recalculate an abstract class's abstraction status. This
   function should be called if a class's abstract methods have been
   implemented or changed after it was created. Usually, this function should
   be called from within a class decorator.

   Returns *cls*, to allow usage as a class decorator.

   If *cls* is not an instance of ABCMeta, does nothing.

>    .. note::
> 
>       This function ignores whatever subclasses *cls* might have. This means
>       that if the decorator itself creates subclasses to *cls*, it must also
>       call *update_abstractmethods* on each of them.
> 
>    .. note::
> 
>       This function assumes that *cls*'s superclasses are already updated.

Reviewer 2: I prefer to do less editorializing.

```suggestion
   .. note::

      This function assumes that *cls*'s superclasses are already updated.
      It does not update any subclasses.
```
